File: src/IntcoEdge.EdgeHost/Models/YkResponseDto.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace IntcoEdge.EdgeHost.Models;

/// <summary>
/// 英科网关统一响应包装体。
/// 反编译来源：`com.hikrobotics.solution.module.yingke.dto.YKResponseDTO`。
/// `Result` 字段在 PSM/Java 端是 `Object`，按需反序列化为强类型。
/// </summary>
public record class YkResponseDto
{
    /// <summary>是否成功（顶层 boolean）。</summary>
    [JsonPropertyName("Success")]
    public bool? Success { get; init; }

    /// <summary>英科网关返回的消息（错误时含原因）。</summary>
    [JsonPropertyName("Message")]
    public string? Message { get; init; }

    /// <summary>结果载荷（PSM 端类型为 Object，按需反序列化）。</summary>
    [JsonPropertyName("Result")]
    public JsonElement? Result { get; init; }

    /// <summary>调用上下文（响应里也带回新 ticket + InvOrgId）。</summary>
    [JsonPropertyName("Context")]
    public YkContextDto? Context { get; init; }

    /// <summary>把 <see cref="Result"/> 反序列化为指定类型（Result 为 null 时返回 default）。</summary>
    public T? DeserializeResult<T>(JsonSerializerOptions? options = null)
    {
        if (Result is null) return default;
        return Result.Value.Deserialize<T>(options ?? IntcoEdge.EdgeHost.Clients.IntcoHttpClient.DefaultJsonOptions);
    }
}

File: DataupLoad/src/main/java/com/hikrobotics/solution/module/yingke/dto/YKResponseDTO.java
package com.hikrobotics.solution.module.yingke.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * yingke MES 响应 DTO（PSM 同名字段命名沿用 1:1）。
 */
public class YKResponseDTO {
   public static YKResponseDTO DEFAULT = new YKResponseDTO().setSuccess(false);
   @JsonProperty("Success")
   private Boolean Success;
   @JsonProperty("Message")
   private String Message;
   @JsonProperty("Result")
   private Object Result;
   @JsonProperty("Context")
   private ContextDTO Context;

   public Boolean getSuccess() {
      return this.Success;
   }

   public String getMessage() {
      return this.Message;
   }

   public Object getResult() {
      return this.Result;
   }

   public ContextDTO getContext() {
      return this.Context;
   }

   @JsonProperty("Success")
   public YKResponseDTO setSuccess(Boolean Success) {
      this.Success = Success;
      return this;
   }

   @JsonProperty("Message")
   public YKResponseDTO setMessage(String Message) {
      this.Message = Message;
      return this;
   }

   @JsonProperty("Result")
   public YKResponseDTO setResult(Object Result) {
      this.Result = Result;
      return this;
   }

   @JsonProperty("Context")
   public YKResponseDTO setContext(ContextDTO Context) {
      this.Context = Context;
      return this;
   }
}
